lib/repository.py

```diff
@@ -81,7 +82,39 @@ def flush(self):
     if graphical:
         splash.Hide()
         sys.stdout = sys.__stdout__
-    print "The retriever is up-to-date"
+    print "\nThe retriever is up-to-date"
+
+
+def update_progressbar(progress):
+    """Shows progressbar given progress in float
+    Example: 0.5 for 50%
+    """
+    # Modify this to change the length of the progress bar
+    barWidth = 20
+    try:
+        rows, columns = os.popen('stty size', 'r').read().split()
+        barWidth = int(columns) - 20
+        if(not (barWidth > 1)):
+            barWidth = 20
+    except:
+        pass
+    status = ""
+    if isinstance(progress, int):
+        progress = float(progress)
```

Comment:
how about we change the logic and assume that given a number just convert to float, and this case we may not need to check if its int.
so before line 102, we can just say ` progress = float(progress)`